tracing/src/span.rs

```diff
@@ -720,10 +720,21 @@ impl Span {
     ///
     /// If this span is disabled, or the resulting follows-from relationship
     /// would be invalid, this function will do nothing.
-    pub fn follows_from(&self, from: impl for<'a> Into<Option<&'a Id>>) -> &Self {
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use tracing::{span, Id, Level, Span};
+    /// # fn main() {
+    /// let span = span!(Level::INFO, "hello!");
+    /// let curr = Span::current();
+    /// span.follows_from(curr.id());
+    /// # }
+    /// ```
+    pub fn follows_from(&self, from: impl Into<Option<Id>>) -> &Self {
```

Comment:
not convinced this is the right type...maybe it should be AsRef or something?

Reply:
Do you mean something like `AsRef<Option<Id>>`?

Reply:
perhaps, although that seems a little strange. It just seems like since we only need an `&Id` to pass into the `Dispatch` fn, we shouldn't have to clone it if the method is called with a ref. but on the other hand an Id is a u64 so I dunno why i care about this